cmake/Findnl-3.cmake

```diff
@@ -0,0 +1,23 @@
+find_package(PkgConfig REQUIRED)
+
+find_library(NL3_LIBRARY "nl-3")
+if(NOT NL3_LIBRARY)
+	message(FATAL_ERROR "libnl-3 not found!")
+endif()
+
+pkg_check_modules(LIBNL3 libnl-3.0 REQUIRED)
```

Comment:
Use *either* `pkg_check_modules` *or* `find_library`. Alternatively, you can use `pkg_check_modules` without the REQUIRED tag and fall back on `find_library` if not found. But that is not very useful, because you can expect the `.pc` file to be installed if the headers are installed.

Reply:
The reason we used both is since we need to set the IMPORTED_LOCATION and INCLUDE_DIRS properties of the imported target - we get the imported location from `find_library` and the include directories from `pkg_check_modules`. How would you do this otherwise?

Reply:
These are very helpful comments and I started revising the Findnl-3.cmake when I realized we don't really use it anymore in Linux mode. Therefore, I suggest we simply delete it, and do the cmake changes for the other find module files correctly in a different commit.

Reply:
Sounds good.